File: NotesApp/ConsoleUI.swift
class ConsoleUI {
    func showCanvas(_ canvas:CanvasUnit){
        print(canvas.drawCanvas())
    }
    
    func readString(_ message: String = "Введите строку") -> String? {
        print(message,terminator: " ")
        return readLine()
    }
    
    func readInt(_ message: String = "Введите число") -> Int? {
        print(message,terminator: " ")
        guard let input = readLine(), let number = Int(input) else {
            return nil
        }
        return number
    }
    
    func showMenuList(_ buttons:[String]) -> Int? {
        print("\n=== МЕНЮ ===")
        for (index,button) in buttons.enumerated(){
            print("\(index+1). \(button)")
        }
        print("================")
        
        return readInt("Выберите пункт меню (1-\(buttons.count)): ")
        
    }
}

File: BazaMobileHW/src/main/NotesApp/src/ConsoleUI.kt
class ConsoleUI {
    fun showCanvas(canvas: CanvasUnit) {
        println(canvas.drawCanvas())
    }

    fun readString(message: String = "Введите строку"): String? {
        print("$message ")
        return readLine()?.takeIf { it.isNotBlank() }
    }

    fun readInt(message: String = "Введите число"): Int? {
        print("$message ")
        return readLine()?.toIntOrNull()
    }

    fun showMenuList(buttons: List<String>): Int? {
        println("\n=== МЕНЮ ===")
        buttons.forEachIndexed { index, button ->
            println("${index + 1}. $button")
        }
        println("================")
        return readInt("Выберите пункт меню (1-${buttons.size})")
    }
}